_posts/2021-02-24-#21224.md

```diff
@@ -0,0 +1,79 @@
+---
+layout: pr
+date: 2021-02-24
+title: "Halt processing of unrequested transactions"
+pr: 21224
+authors: [ariard]
+components: ["p2p"]
+host: ellemouton
+status: upcoming
+commit:
+---
+
+## Notes
+
+- Transaction relay is a three step process: `inv` -> `getdata` -> `tx`:
+
+  1. the relaying node sends an `inv` message to the receiving node to announce
+     a new transaction.
+  2. the receiving node sends a `getdata` message to the relaying node to
+     request the full transaction.
+  3. the relaying node delivers a `tx` message to the receiving node. If the
+     relaying node is no longer able to deliver the transaction, it responds
+     with `notfound` instead of the `tx`.
+
+  You can learn more about these messages [here](https://en.bitcoin.it/wiki/Protocol_documentation#inv).
+
+- Currently if a `tx` message is received it will not be processed if:
+
+	1. The node is in blocks-only mode and the peer sending the message has 
+	no relay permissions
+	2. The peer is a blocks-relay only peer
+
+  In all other cases, the node will process a transaction that it receives even 
```

Comment:
It depends what you mean by 'process' here. We'll do the `AlreadyHaveTx()` checks first, to make sure that we don't process the same transaction more than once: https://github.com/bitcoin/bitcoin/blob/9f3ffa29389aa59ea87b5aa974a2da17ccfd31a4/src/net_processing.cpp#L1619-L1648.